src/app/games/trexo/TrexoMove.ts

```diff
@@ -0,0 +1,46 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { MoveCoordToCoord } from 'src/app/jscaip/MoveCoordToCoord';
+import { NumberEncoder } from 'src/app/utils/Encoder';
+import { Localized } from 'src/app/utils/LocaleUtils';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { TrexoState } from './TrexoState';
+
+export class TrexoMoveFailure {
+
+    public static readonly NON_NEIGHBOR_COORDS: Localized = () => $localize`Thoses two coord are not neighbors!`;
+}
+
+/**
+ * It's not really a move coord to coord but rather a "drop two coords" (that are neighbor)
+ */
+export class TrexoMove extends MoveCoordToCoord {
+
+    public static encoder: NumberEncoder<TrexoMove> =
+        MoveCoordToCoord.getEncoder<TrexoMove>(TrexoState.SIZE, TrexoState.SIZE,
+                                               (zero: Coord, one: Coord): TrexoMove => {
+                                                   return TrexoMove.from(zero, one).get();
+                                               });
+
+    public static from(zero: Coord, one: Coord): MGPFallible<TrexoMove> {
+        if (zero.isNotInRange(TrexoState.SIZE, TrexoState.SIZE) ||
+            one.isNotInRange(TrexoState.SIZE, TrexoState.SIZE))
+        {
+            return MGPFallible.failure('TODOTODO OUT OF RANGE FROM Lasca');
+        }
+        const distance: number = zero.getOrthogonalDistance(one);
+        if (distance === 1) {
+            return MGPFallible.success(new TrexoMove(zero, one));
+        } else {
+            return MGPFallible.failure(TrexoMoveFailure.NON_NEIGHBOR_COORDS());
+        }
+    }
+    private constructor(public readonly zero: Coord, public readonly one: Coord) {
```

Comment:
is zero/one really better than coord0/coord1 ? :D

Reply:
ça voulait dire "qui appartient au joueur zéro" et "qui appartient au joueur un" donc ouais c'était p-e trop court mais carrément mieux !

Tiens, comment "renommer" le nom d'un champ hérité d'un parent ?

Reply:
renommer dans quel sens ? Ce que tu fais maintenant c'est bien (prendre deux arguments passés au constructeur du parent).  Sinon tu peux les copier, mais bof.

Reply:
renommer dans le sens que MoveTwoCoord ~= { first, second }
MoveCoordToCoord = { start, end }
TrexoCoord = { pieceZero, pieceOne }
J'imagine que le rendre privé à la mère et que même les enfants doivent utiliser un accesseseur, et mieux vaut dupliquer les accesseurs que les champs ?

Reply:
oui à tout !

Reply:
fiou, ça a encore changé pas mal de brins de code ! mais moveCoordToCoord qui move "coord" vers "end" ça ne nous manquera pas haha :D !